File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletMultiTouchV2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletMultiTouchV2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Multi Touch Bricklet 2.0

// Callback function for touch state callback
function cb_touchState($state)
{
    echo "Electrode 0: $state\n";
    echo "Electrode 1: $state\n";
    echo "Electrode 2: $state\n";
    echo "Electrode 3: $state\n";
    echo "Electrode 4: $state\n";
    echo "Electrode 5: $state\n";
    echo "Electrode 6: $state\n";
    echo "Electrode 7: $state\n";
    echo "Electrode 8: $state\n";
    echo "Electrode 9: $state\n";
    echo "Electrode 10: $state\n";
    echo "Electrode 11: $state\n";
    echo "Proximity: $state\n";
    echo "\n";
}

$ipcon = new IPConnection(); // Create IP connection
$mt = new BrickletMultiTouchV2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register touch state callback to function cb_touchState
$mt->registerCallback(BrickletMultiTouchV2::CALLBACK_TOUCH_STATE, 'cb_touchState');

// Set period for touch state callback to 0.01s (10ms)
$mt->setTouchStateCallbackConfiguration(10, TRUE);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/go/example_callback.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/multi_touch_v2_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Multi Touch Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	mt, _ := multi_touch_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	mt.RegisterTouchStateCallback(func(state [13]bool) {
		fmt.Printf("Electrode 0: %s\n", state[0])
		fmt.Printf("Electrode 1: %s\n", state[1])
		fmt.Printf("Electrode 2: %s\n", state[2])
		fmt.Printf("Electrode 3: %s\n", state[3])
		fmt.Printf("Electrode 4: %s\n", state[4])
		fmt.Printf("Electrode 5: %s\n", state[5])
		fmt.Printf("Electrode 6: %s\n", state[6])
		fmt.Printf("Electrode 7: %s\n", state[7])
		fmt.Printf("Electrode 8: %s\n", state[8])
		fmt.Printf("Electrode 9: %s\n", state[9])
		fmt.Printf("Electrode 10: %s\n", state[10])
		fmt.Printf("Electrode 11: %s\n", state[11])
		fmt.Printf("Proximity: %s\n", state[12])
		fmt.Println()
	})

	// Set period for touch state callback to 0.01s (10ms).
	mt.SetTouchStateCallbackConfiguration(10, true)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
